Refactor/robot motion observer

martiniil opened the pull request:
A part of the integrationtest utility functions is replaced by the new test utility class `RobotMotionObserver`. A `RobotMotionObserver` object can detect, if the robot is moving and wait for a movement to start or to stop.

martiniil committed:
Add RobotMotionObserver in testutils

martiniil committed:
Remove deprecated testutils

martiniil committed:
Fix acceptance test

agutenkunst commented on pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py:
```diff
@@ -0,0 +1,115 @@
+    """
+
+    _DEFAULT_WAIT_TIME_FOR_MOTION_DETECTION = 3.0
+    _DEFAULT_TOLERANCE_FOR_MOTION_DETECTION = 0.01
```
I think the unit is missing.

martiniil replied:
done.

agutenkunst commented on pilz_industrial_motion_testutils/src/pilz_industrial_motion_testutils/robot_motion_observer.py:
```diff
@@ -0,0 +1,115 @@
+                break
+
+            if (rospy.get_time() - start_time > wait_time_out):
+                rospy.loginfo("Reached timeout when waiting for motion start.")
```
To be really precise you should set `motion_started = False` to (better) cover edge cases.

martiniil replied:
not necessary.

agutenkunst commented on pilz_robot_programming/test/integrationtests/tst_api_gripper.py:
```diff
@@ -33,6 +34,7 @@ class TestAPIGripper(unittest.TestCase):
 
     def setUp(self):
         self.robot = Robot(API_VERSION)
+        self.robot_motion_observer = RobotMotionObserver("gripper")
```
You could work with `RobotMotionObserver(group_name = "gripper") to be move expressiv.

martiniil replied:
done.

agutenkunst reviewed (changes requested):
Just minor change requests...

martiniil committed:
Minor changes from review

martiniil committed:
Make ctrl+c test more robust

In this test a motion is started usind movecmd, which has more initialization than the usual r.move().
So the waiting time for the motion to start is increased.
